python-client/giskard/scanner/performance.py

```diff
@@ -0,0 +1,235 @@
+
+        return issues
+
+    def _get_test_name(self, test):
+        # @TODO: throw this away when we can
+        if test.meta.name.endswith("f1"):
+            return "F1 score"
+        if test.meta.name.endswith("accuracy"):
+            return "Accuracy"
+        if test.meta.name.endswith("recall"):
+            return "Recall"
+        if test.meta.name.endswith("rmse"):
+            return "RMSE"
+        if test.meta.name.endswith("precision"):
+            return "Precision"
+
+    def _diff_test(self, slice_fn, model, dataset, test_fn, threshold):
+        # Apply the test
+        test = test_fn(
+            actual_dataset=dataset.slice(slice_fn),
+            reference_dataset=dataset,  # Could exclude slice_dataset for independence
+            model=model,
+            threshold=threshold,
+            direction=Direction.Increasing,
+        )
+
+        res = test.execute()
+
+        return res
+
+    def _calculate_meta(self, model, dataset):
+        true_target = dataset.df.loc[:, dataset.target].values
+        pred = model.predict(dataset)
+
+        loss_values = [
+            metrics.log_loss([true_label], [probs], labels=model.meta.classification_labels)
+            for true_label, probs in zip(true_target, pred.raw)
+        ]
+
+        return pd.DataFrame({"__gsk__loss": loss_values}, index=dataset.df.index)
+
+    def _find_slices(self, dataset, model, meta: pd.DataFrame, slicer_name):
+        df_with_meta = dataset.df.join(meta)
+        target_col = "__gsk__loss"
+
+        # @TODO: Handle this properly once we have support for metadata in datasets
+        column_types = dataset.column_types.copy()
+        column_types["__gsk__loss"] = "numeric"
+        dataset_with_meta = Dataset(df_with_meta, target=dataset.target, column_types=column_types)
+
+        # Columns by type
+        cols_by_type = {
+            type_val: [col for col, col_type in dataset.column_types.items() if col_type == type_val]
+            for type_val in ["numeric", "category", "text"]
+        }
+
+        # Numerical features
+        slicer = get_slicer(slicer_name, dataset_with_meta, target_col)
+
+        slices = []
+        for col in cols_by_type["numeric"]:
+            slices.extend(slicer.find_slices([col]))
+
+        # Categorical features
+        slicer = CategorySlicer(dataset_with_meta, target=target_col)
+        for col in cols_by_type["category"]:
+            slices.extend(slicer.find_slices([col]))
+
+        # @TODO: FIX THIS
+        # Text features
+        slicer = TextSlicer(dataset_with_meta, target=target_col, slicer=slicer_name)
+        for col in cols_by_type["text"]:
+            slices.extend(slicer.find_slices([col]))
+
+        # @TODO: Should probably bind back to original dataset, but how to to pass
+        # cleanly also the metadata needed for the plots? Maybe need a wrapper object
+        # like Issue or similar.
```

Comment:
sounds like a good idea, something like `DatasetSlices` class that combines the two objects: dataset and found slices?

Reply:
Actually this has been done but the comment stayed ;) It’s the `Issue` class.